pkg/kp/rc.go

```diff
@@ -0,0 +1,30 @@
+package kp
+
+import (
+	"github.com/square/p2/pkg/labels"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/rc"
+)
+
+// ReplicationControllerStore represents an interface for persisting replication controllers to Consul,
+// as well as restoring replication controllers from Consul.
+type ReplicationControllerStore interface {
+	// Create creates a replication controller with the specified manifest and selectors.
+	// The node selector is used to determine what nodes the replication controller may schedule on.
+	// The pod label set is:
+	// 1) applied to every pod the replication controller schedules,
+	// 2) used (in addition to replication controller ID) to determine what pods are already scheduled.
+	Create(manifest pods.Manifest, nodeSelector labels.Selector, podLabels labels.Set) (ReplicationControllerHandle, error)
+
+	// List lists all of the replication controllers.
+	List() ([]ReplicationControllerHandle, error)
```

Comment:
What makes a controller "active"?

---

I assume that the data in Consul will allow:
1. Querying for the status of a deployment as it's happening.
2. Resuming a deployment if one node happens to crash.

For (1), it seems reasonable to want status from a specific deployment, but I wouldn't bother with it until we have direct use case (e.g., until said UI is built).  For (1) and (2), you need discovery, and `List()` is probably sufficient for a first pass.

But also, to support (2) in the presence of many orchestrator instances, do we need some kind of locking on each request?